### DISCUSSION TITLE
Cant recieve requests when one thread blocked on windows

### DISCUSSION THREAD

**Discussion by EgorWeders** [cutelyst/cutelyst#452] [Q&A] [answered]
I have 4 working cores /apps  and a controllers. When I block a thread (waitForReady, QThread::sleep(),loops and etc.) on windows, I can't get new contexts. On Linux, I can get them from other Cutelsyt::applications/workers as long as all available threads aren't blocked. A blocked thread is not the same as an engine's or server's thread, it just uses only the latest forked worker.

**Comment by dantti** [comment, marked as answer]
There is a `--experimental-thread-balancer` which iirc has a single thread receiving new connections and forwarding to threads, I don't use Windows so dunno why it doesn't just do what POSIX does and notifies all threads about new connections (unless `--reuse-port` is set).
Still if you can avoid blocking code that is then much better.

**Reply by EgorWeders** [reply]
It cannot be avoided due some IPC, but ty for answer.

**Reply by dantti** [reply]
Well you can avoid it by creating a thread to deal with this IPC and async wait for the result.
